types/priv_validator.go

```diff
@@ -225,7 +227,14 @@ func (privVal *PrivValidatorFS) save() {
 		// `@; BOOM!!!
 		cmn.PanicCrisis(err)
 	}
-	err = cmn.WriteFileAtomic(privVal.filePath, jsonBytes, 0600)
+	outFile := privVal.filePath
+	// We need to ensure that the parent directory exists.
+	// See Issue https://github.com/tendermint/tendermint/issues/1290
+	dir := filepath.Dir(outFile)
+	if err := os.MkdirAll(dir, 0700); err != nil {
```

Comment:
Should we move this into `cmn.WriteFileAtomic`? The flow is

1 <...> - 2 <priv_validator.go> - 3 <common.go>

So we should either create the directory at (1) or (3). (2) feels like a halfway.

Reply:
:+1:

Reply:
Great point @melekes, in that case I'd rather do it in 3) cmn.WriteFileAtomic. I didn't do so initially because I assumed changing the implementation of cmn.WriteFileAtomic was off-limits.

Reply:
Okay now I recall why I put this here, doing MkdirAll in cmn.WriteFileAtomic is quite complex:
Atomic meaning "all or nothing" would require that I have to check the:
a) The least none existent path e.g if we have a/b/c/d/e/f/g/txt.json, if directory b doesn't exist, I'll have to find out firstly what doesn't exist to clean it up if the operation fails
b) Remove the least existent path

I am not sure I'd want to be introducing such  a change in common at a time like this close to launch and moreover when my dev bandwidth is running out. For those reasons, I'll leave this here, someone else can
later move that check into WriteFileAtomic since that had never come up anyways.

Reply:
ok, I tend to agree